gating/check/pre_deploy.sh

```diff
@@ -60,8 +60,19 @@ if [[ ! -e phantomjs-2.1.1-linux-x86_64.tar.bz2 ]]; then
   tar -xjf phantomjs-2.1.1-linux-x86_64.tar.bz2
 fi
 
+# FrankZhang: Limit venv python to python2 
+# to avoid python3 code making syntax error https://rpc-openstack.atlassian.net/browse/RI-458
 if [[ ! -d ".venv" ]]; then
-  virtualenv .venv
+  virtualenv .venv --python=python2 --always-copy --no-download
+fi
+
+# FrankZhang: Upgrade pip and setuptools due to outdate issue https://rpc-openstack.atlassian.net/browse/RI-458
+if [[ $(pip -V | awk '{print $2}') < 9.0.1 ]]; then 
+    pip install --upgrade pip==9.0.1
+fi
+
+if [[ $(pip list | grep setuptools | awk '{print $2}') < 33.1.1 ]]; then
+    pip install --upgrade setuptools==33.1.1
```

Comment:
I suspect that bashate is getting confused through the syntax here. It might be worth doing something like this?

```
pip_version=$(pip -V | awk '{print $2}')
if [[ ${pip_version} < 9.0.1 ]]; then
```

Or you could try reading the code for it to figure out what's going on.